### ISSUE TITLE
v-bind:key required when iterable is a list

### ISSUE THREAD

**Issue by BlueskyFR** [vuejs/eslint-plugin-vue#2064]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.30.0
- **eslint-plugin-vue version:** 9.8.0
- **Node version:** v18.12.1
- **Operating System:** Arch Linux

**Please show your full configuration:**

```js
/* eslint-env node */
require("@rushstack/eslint-patch/modern-module-resolution");

module.exports = {
  root: true,
  extends: [
    "plugin:vue/vue3-essential",
    "eslint:recommended",
    "@vue/eslint-config-typescript",
    "@vue/eslint-config-prettier",
  ],
  parserOptions: {
    ecmaVersion: "latest",
  },
  parser: "vue-eslint-parser",
};
```

**What did you do?**	

```vue
<template>
  <p v-for="i in [1, 2]">{{ i }}</p>
</template>
```

**What did you expect to happen?**
No error expecting any `v-bind:key` attribute on a list.

**What actually happened?**

![image](https://user-images.githubusercontent.com/25421460/209016710-d3e1bb72-0078-42bb-9699-a4e4ad70db76.png)
```
  1:1  error  Component name "aa" should always be multi-word               vue/multi-word-component-names
  2:3  error  Elements in iteration expect to have 'v-bind:key' directives  vue/require-v-for-key

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by ota-meshi** [comment]
I think the rule is working correctly because you didn't specify a `v-bind:key`. Turn off the rule if you don't want to specify `v-bind:key`.

**Comment by BlueskyFR** [comment]
@ota-meshi how can you specify a key when the iterable is a list like this?

**Comment by FloEdelmann** [comment]
`:key="i"`

**Comment by BlueskyFR** [comment]
> `:key="i"`

Yes but what if elements are not unique?

**Comment by FloEdelmann** [comment]
You *could* use the array index as the key, but only if you are sure that the array is basically read-only:

```vue
<p v-for="(item, index) in [7, 5, 7]" :key="index">{{ item }}</p>
```
